app/assets/javascripts/discourse/app/components/sidebar/user/custom-sections.js

```diff
@@ -7,6 +7,136 @@ import { ajax } from "discourse/lib/ajax";
 import { iconHTML } from "discourse-common/lib/icon-library";
 import { htmlSafe } from "@ember/template";
 import { bind } from "discourse-common/utils/decorators";
+import { tracked } from "@glimmer/tracking";
+
+class Section {
+  @tracked dragCss;
+  @tracked links;
+
+  constructor(section, currentUser, router) {
+    this.section = section;
+    this.router = router;
+    this.currentUser = currentUser;
+    this.slug = section.slug;
+
+    this.links = this.section.links.map((link) => {
+      return new SectionLink(link, this, this.router);
+    });
+  }
+
+  get decoratedTitle() {
+    return this.section.public && this.currentUser.staff
+      ? htmlSafe(`${iconHTML("globe")} ${this.section.title}`)
+      : this.section.title;
+  }
+
+  get headerActions() {
+    if (!this.section.public || this.currentUser.staff) {
+      return [
+        {
+          action: () => {
+            return showModal("sidebar-section-form", { model: this.section });
+          },
+          title: I18n.t("sidebar.sections.custom.edit"),
+        },
+      ];
+    }
+  }
+
+  @bind
+  disable() {
+    this.dragCss = "disabled";
+  }
+
+  @bind
+  enable() {
+    this.dragCss = null;
+  }
+
+  @bind
+  moveLinkDown(link) {
+    const position = this.links.indexOf(link) + 1;
+    this.links = this.links.removeObject(link);
+    this.links.splice(position, 0, link);
+  }
+
+  @bind
+  moveLinkUp(link) {
+    const position = this.links.indexOf(link) - 1;
+    this.links = this.links.removeObject(link);
+    this.links.splice(position, 0, link);
+  }
+  @bind
+  reorder() {
+    return ajax(`/sidebar_sections/reorder`, {
+      type: "POST",
+      contentType: "application/json",
+      dataType: "json",
+      data: JSON.stringify({
+        sidebar_section_id: this.section.id,
+        links_order: this.links.map((link) => link.id),
+      }),
+    });
+  }
+}
+
+class SectionLink {
```

Comment:
Same for https://github.com/discourse/discourse/pull/20626/files#r1141590452 as well. Let's split it into a file of its own.